web/src/js/ducks/modes/reverse.ts

```diff
@@ -0,0 +1,140 @@
+import {
+    RECEIVE as RECEIVE_STATE,
+    UPDATE as UPDATE_STATE,
+} from "../backendState";
+import {
+    ModeState,
+    getModesOfType,
+    includeModeState,
+    updateMode,
+} from "./utils";
+import type { ModesState } from "../modes";
+
+export const MODE_REVERSE_TOGGLE = "MODE_REVERSE_TOGGLE";
+export const MODE_REVERSE_SET_LISTEN_CONFIG = "MODE_REVERSE_SET_LISTEN_CONFIG";
+export const MODE_REVERSE_SET_DESTINATION = "MODE_REVERSE_SET_DESTINATION";
+export const MODE_REVERSE_SET_PROTOCOL = "MODE_REVERSE_SET_PROTOCOL";
+export const MODE_REVERSE_ERROR = "MODE_REVERSE_ERROR";
+
+export interface ReverseState extends ModeState {
+    protocol?: string;
```

Comment:
This can be typed more strongly. :)

Reply:
you mean that the protocol should be one of these: `"https","http", "dns" etc...`